ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/CustodyGroupCountManagerImpl.java

```diff
@@ -148,6 +156,42 @@ public void setCustodyGroupSyncedCount(final int custodyGroupSyncedCount) {
     custodyGroupSyncedCountGauge.set(custodyGroupSyncedCount);
   }
 
+  private void checkAndHandleGenesisTransition(final Set<UInt64> nodeValidatorIndices) {
+    combinedChainDataClient
+        .getLatestFinalized()
+        .ifPresent(
+            finalizedAnchor -> {
+              // Check if we've moved beyond genesis epoch (epoch 0)
+              if (finalizedAnchor.getEpoch().isGreaterThan(UInt64.ZERO)) {
+                // We've finalized beyond genesis, now get the genesis state to count validators
+                final UInt64 genesisSlot = spec.computeStartSlotAtEpoch(UInt64.ZERO);
```

Comment:
This to me look a little redundant but I could be wrong, you can probably assume that genesis always start at slot 0 can't you? There might be a world where we have a network with genesis not starting at slot 0.
Also I need to check if it's safe to do this because in case of a restart you might be way ahead of genesis and still setting cgc based on genesis.